### PR TITLE
Skip username validation if the user already exists in email username–enabled scenario

### PR DESCRIPTION
**Pull request by DilshanSenarath**
### Related Issue
- https://github.com/wso2/product-is/issues/27633

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling when verifying user existence during account provisioning and password recovery operations.
  * Enhanced username validation logic to properly handle non-email usernames when users already exist.

* **Tests**
  * Added new test scenarios for user existence checks and username validation edge cases.

### CODE REVIEW

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/executor/UserProvisioningExecutor.java [hunk 1]
```diff
@@ -329,12 +332,30 @@ private String resolveUsername(FlowUser user, String tenantDomain) throws FlowEn
...
+                    : resolveUserStoreDomain(user.getUsername());
+            UserStoreManager userStoreManager = getUserStoreManager(context.getTenantDomain(), userStoreDomain,
+                    context.getContextIdentifier(), context.getFlowType());
+            return userStoreManager.isExistingUser(user.getUsername());
+        } catch (UserStoreException e) {
+            String maskedUsername = LoggerUtils.isLogMaskingEnable
+                    ? LoggerUtils.getMaskedContent(user.getUsername()) : user.getUsername();
+            throw handleServerException(ERROR_CODE_USER_EXISTENCE_CHECK_FAILURE, e,
+                    maskedUsername, context.getContextIdentifier());
+        }
```
[the remark this answers is not in the document]
Comment (DilshanSenarath) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/executor/UserProvisioningExecutor.java:356]: These are same values

**Review by ZiyamSanthosh — APPROVED** [review]
